ed-odyssey-materials-helper: init at 2.156

elfenermarcell opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->
https://github.com/jixxed/ed-odyssey-materials-helper
ed-odyssey-materials-helper is a utility for managing materials in the game Elite Dangerous Odyssey
## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

elfenermarcell committed:
maintainers: add elfenermarcell

elfenermarcell committed:
ed-odyssey-materials-helper: init at 2.156

ToasterUwU commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 389035`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>ed-odyssey-materials-helper</li>
  </ul>
</details>

ToasterUwU reviewed (changes requested):
Just noticed something actually. This needs `desktop-file-utils` at runtime to register itself as a url handler for edomh:// type links. Without it fails to register itself properly. 

I tested it with `desktop-file-utils` and that works nicely. So please add `desktop-file-utils` as a runtime dependency.

ToasterUwU commented:
Since this needs a touchup anyways, might as well update this to the newest version. Dont need to if thats too much rn, but would be appreciated.

elfenermarcell commented:
> Just noticed something actually. This needs `desktop-file-utils` at runtime to register itself as a url handler for edomh:// type links. Without it fails to register itself properly.
> 
> I tested it with `desktop-file-utils` and that works nicely. So please add `desktop-file-utils` as a runtime dependency.

The problem is, that option in the settings creates a desktop file in your home dir, containing the full path to the non-wrapped executable, store path and all. That means after an update, the url handling would stop working.

The "proper" solution I think is to just add `mimeTypes = [ "x-scheme-handler/edomh" ];` to `makeDesktopItem` - of course then this option in the settings menu is useless/potentially bad as it overrides the desktop file in the package.

I'll see if it can be patched out.

ToasterUwU commented:
Sounds good.

I agree that setting the handler in the package makes the most sense. If you need any help or something let me know. Otherwise I will wait and review again when something changed.

elfenermarcell commented:
@ToasterUwU I added you as a maintainer.

I've made some changes and added a patch which should remove the url scheme settings, it is just set in the desktop file now. Also updated to the latest version and added wrapGAppsHook3 to make the file dialogs open reliably.

elfenermarcell commented:
Actually it looks like that although this makes the edomh: URL work with `xdg-open` it doesn't with Firefox (`Prevented navigation to [...] due to an unknown protocol.`) Does that mean running `xdg-mime default` is still neccesary?

ToasterUwU commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 389035`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>ed-odyssey-materials-helper</li>
  </ul>
</details>

ToasterUwU commented:
@elfenermarcell At first i also thought it works for firefox. But then i deleted the "edomh.desktop" from .local/share/applications  which was a leftover from the previous one, and it didnt work. xdg-open works for me as well, but not firefox.

ToasterUwU reviewed (changes requested):
Need to find a solution to have firefox also open the program, not just xdg-open

elfenermarcell commented:
So I tested it by adding it into my system packages (both with my actual nixos install and a build-vm one) and the URL scheme works just fine.

I think this problem is caused by testing it with just `nix-env -i ./result` (which is how I tried to test it first). In fact, I can also replicate this with the freetube package. Adding it to my environment with nix-env adds the desktop file but the freetube: URL scheme doesn't work in Firefox. But if I add it to my system packages, it does work. It looks like nix-env doesn't do something that is required for this.

ToasterUwU commented:
I will test it as well like that, i used nixpkgs-review before and started firefox from within the shell it makes.

ToasterUwU reviewed (approved):
Can confirm, when installing properly, the handler is working nicely. 

This is ready to merge!

Scrumplex commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 389035`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>ed-odyssey-materials-helper</li>
  </ul>
</details>

Scrumplex commented:
https://github.com/jixxed/ed-odyssey-materials-helper/releases/tag/2.156

Scrumplex commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 389035`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>ed-odyssey-materials-helper</li>
  </ul>
</details>
